cpp/src/Ice/SSL/SchannelTransceiverI.cpp

```diff
@@ -874,10 +925,31 @@ Schannel::TransceiverI::TransceiverI(
       _peerCertificate(nullptr),
       _remoteCertificateValidationCallback(clientAuthenticationOptions.serverCertificateValidationCallback),
       _certificate(nullptr),
-      _rootStore(nullptr),
+      _rootStore(clientAuthenticationOptions.trustedRootCertificates),
       _credentials({0, 0}),
-      _ssl({0, 0})
+      _ssl({0, 0}),
+      _chainEngine(nullptr)
 {
+    if (_rootStore && !_remoteCertificateValidationCallback)
```

Comment:
Schannel automatically validate server certificate using the system trusted CAs. If the client doesn't specify a validation callback or custom trusted roots we can use the default validation mechanism.